src/SequentialEventsModule.ts

```diff
@@ -21,30 +21,29 @@ export class SequentialEventsModule implements OnApplicationBootstrap {
   ) {}
 
   public onApplicationBootstrap() {
-    const modules = [...this.modulesContainer.values()];
-    const listenerDefinitions = modules.flatMap(module => {
-      const providers = [...module.providers.values()];
-      const listeners = providers.map(provider => {
-        const instance: any = provider.instance;
-        if (!instance || !instance.constructor) {
-          return undefined;
-        }
-        const listenedEvent = Reflect.getMetadata(SEQUENTIAL_EVENT_LISTENER, instance.constructor);
-        if (!listenedEvent && isEvent(listenedEvent) === false) {
-          return undefined;
-        }
-        return { providerConstructor: instance.constructor, eventType: listenedEvent.type };
+    const listenerDefinitions = [...this.modulesContainer.values()]
+      .flatMap((module) => [...module.providers.values()])
+      .filter((provider) => (provider.instance as any)?.constructor)
+      .filter((provider) =>
+        getListenedEvent((provider.instance as any).constructor),
+      )
+      .map((provider) => {
+        const instance = provider.instance as any;
+
+        const listenedEvent = getListenedEvent(instance.constructor)!;
+
+        return {
+          providerConstructor: instance.constructor,
+          eventType: listenedEvent.type,
+        } as ListenerDefinition;
       });
-      return listeners.filter(Boolean);
-    }) as ListenerDefinition[];
 
-    const listenerParis = listenerDefinitions.map(definition => {
-      const instance = this.moduleRef.get(definition.providerConstructor, { strict: false });
-      return { listener: instance, eventType: definition.eventType };
-    });
+    listenerDefinitions.forEach((definition) => {
+      const instance = this.moduleRef.get(definition.providerConstructor, {
+        strict: false,
+      });
 
-    listenerParis.forEach(pair => {
-      this.eventBus.register(pair.listener, pair.eventType);
+      this.eventBus.register(instance, definition.eventType);
```

Comment:
I refactored it a little bit to be able to unit test it, but in the end it looks like there is still too much to be stubbed from Nest.js modules side. I implemented e2e tests instead.